File: 2634-minimum-common-value/minimum-common-value.py
class Solution(object):
    def getCommon(self, nums1, nums2):
        p1, p2, n, m = 0, 0, len(nums1), len(nums2)
        while p1 < n and p2 < m: 
            if nums1[p1] == nums2[p2]:
                return nums1[p1]
            elif nums1[p1] > nums2[p2]:
                p2 += 1
            else:
                p1 += 1
        return -1
        

File: 2634-minimum-common-value/minimum-common-value.cpp
class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int p1 = 0, p2 = 0, n = nums1.size(), m = nums2.size();
        while(p1 < n && p2 < m) {
            if(nums1[p1] == nums2[p2]) return nums1[p1];
            else if(nums1[p1] > nums2[p2]) p2++;
            else p1++;
        }
        return -1;
    }
};